From 19b8662207ca1e515eccb8fd0b1edcb49b62700b Mon Sep 17 00:00:00 2001
From: "Tom G. Christensen" <tgc@jupiterrise.com>
Date: Thu, 30 Mar 2017 18:12:42 +0200
Subject: [PATCH] Treat empty codeset value as ascii

On Solaris 2.6 locales using an ascii codeset returns an empty value
from nl_langinfo(CODESET) instead of the value '646' as seen in later
Solaris releases.
---
 utf8.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/utf8.c b/utf8.c
index dead79b8..9d63d1d6 100644
--- a/utf8.c
+++ b/utf8.c
@@ -61,7 +61,8 @@ dangerous_locale(void) {
 
 	loc = nl_langinfo(CODESET);
 	return strcmp(loc, "US-ASCII") != 0 && strcmp(loc, "UTF-8") != 0 &&
-	    strcmp(loc, "ANSI_X3.4-1968") != 0 && strcmp(loc, "646") != 0;
+	    strcmp(loc, "ANSI_X3.4-1968") != 0 && strcmp(loc, "646") != 0 &&
+	    strcmp(loc, "") != 0;
 }
 
 static int
-- 
2.11.0

